internal/services/namedlocations/named_location_resource_msgraph.go

```diff
@@ -0,0 +1,225 @@
+		}
+
+	}
+	if v, ok := d.GetOk("country"); ok {
+		properties := &msgraph.CountryNamedLocation{}
+		if d.HasChange("ip") {
+			properties = expandCountryNamedLocation(v.([]interface{}))
+		}
+		properties.BaseNamedLocation = &base
+
+		if _, err := client.UpdateCountry(ctx, *properties); err != nil {
+			return tf.ErrorDiagF(err, "Could not update named location with ID: %q", d.Id())
+		}
+
+	}
+
+	return nil
+}
+
+func namedLocationResourceReadMsGraph(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*clients.Client).NamedLocations.MsClient
+
+	if _, ok := d.GetOk("ip"); ok {
+		location, status, err := client.GetIP(ctx, d.Id())
+		if err != nil {
+			if status == http.StatusNotFound {
+				log.Printf("[DEBUG] Named Location with Object ID %q was not found - removing from state", d.Id())
+				d.SetId("")
+				return nil
+			}
+
+			return tf.ErrorDiagPathF(err, "id", "Retrieving Named Location with object ID %q", d.Id())
+		}
+
+		tf.Set(d, "display_name", location.DisplayName)
+		tf.Set(d, "id", location.ID)
+
+	}
+
+	if _, ok := d.GetOk("country"); ok {
+		location, status, err := client.GetCountry(ctx, d.Id())
+		if err != nil {
+			if status == http.StatusNotFound {
+				log.Printf("[DEBUG] Named Location with Object ID %q was not found - removing from state", d.Id())
+				d.SetId("")
+				return nil
+			}
+
+			return tf.ErrorDiagPathF(err, "id", "Retrieving Named Location with object ID %q", d.Id())
+		}
+
+		tf.Set(d, "display_name", location.DisplayName)
+		tf.Set(d, "id", location.ID)
+
+	}
+
+	return nil
+}
+
+func namedLocationResourceDeleteMsGraph(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*clients.Client).NamedLocations.MsClient
+
+	if _, ok := d.GetOk("ip"); ok {
+		_, status, err := client.GetIP(ctx, d.Id())
+		if err != nil {
+			if status == http.StatusNotFound {
+				log.Printf("[DEBUG] Named Location with ID %q already deleted", d.Id())
+				return nil
+			}
+
+			return tf.ErrorDiagPathF(err, "id", "Retrieving named location with ID %q", d.Id())
+		}
+	}
+
+	if _, ok := d.GetOk("country"); ok {
+		_, status, err := client.GetCountry(ctx, d.Id())
+		if err != nil {
+			if status == http.StatusNotFound {
+				log.Printf("[DEBUG] Named Location with ID %q already deleted", d.Id())
+				return nil
+			}
+
+			return tf.ErrorDiagPathF(err, "id", "Retrieving named location with ID %q", d.Id())
+		}
+	}
+
+	status, err := client.Delete(ctx, d.Id())
+	if err != nil {
+		return tf.ErrorDiagPathF(err, "id", "Deleting named location with ID %q, got status %d", d.Id(), status)
+	}
+
+	return nil
+}
+
+func expandIPNamedLocation(in []interface{}) *msgraph.IPNamedLocation {
+	if len(in) == 0 {
+		return nil
+	}
+
+	result := msgraph.IPNamedLocation{}
+	config := in[0].(map[string]interface{})
+
+	ipRanges := config["ip_ranges"].([]interface{})
+	isTrusted := config["is_trusted"].(interface{})
```

Comment:
This will already be returned as type `interface{}` so no need for the assertion here

Reply:
I've removed the last assertion there but left in the slice of interface assertion as that's the only way I could get the expand function to work
If there's a cleaner way to do it let me know and I'll give it another shot